test/e2e/specs/editor/various/keep-styles-on-block-transforms.spec.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * WordPress dependencies
+ */
+const { test, expect } = require( '@wordpress/e2e-test-utils-playwright' );
+
+test.describe( 'Keep styles on block transforms', () => {
+	test.beforeEach( async ( { admin } ) => {
+		await admin.createNewPost();
+	} );
+
+	test( 'Should keep colors during a transform', async ( {
+		page,
+		editor,
+	} ) => {
+		await page.click( 'role=button[name="Add default block"i]' );
+		await page.keyboard.type( '## Heading' );
+		await editor.openDocumentSettingsSidebar();
+		await page.click( 'role=button[name="Text"i]' );
+		await page.click( 'role=button[name="Color: Luminous vivid orange"i]' );
+
+		await page.click( 'role=button[name="Heading"i]' );
+		await page.click( 'role=menuitem[name="Paragraph"i]' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: { content: 'Heading' },
```

Comment:
The important part of this test is about the color, we should add it to the assertion:

```suggestion
				attributes: { content: 'Heading', textColor: 'luminous-vivid-orange' },
```

Reply:
Feedbacks addressed